scripts/utils.sh

```diff
@@ -0,0 +1,384 @@
+
+  utils::echo_action "Updating ${FSTAB_FILE_PATH} file to automount the data disk using its UUID..."
+  if grep -q "${data_disk_file_system_uuid}" "${FSTAB_FILE_PATH}"; then
+    utils::echo_info "Skipped: already set up."
+  else
+    printf "UUID=${data_disk_file_system_uuid}\t${data_disk_mount_point_path}\t${data_disk_file_system_type}\tdefaults,nofail\t0\t2\n" >> "${FSTAB_FILE_PATH}"
+    utils::echo_info "Done."
+  fi
+
+  utils::echo_action 'Mounting all drives...'
+  mount -a
+  utils::echo_info "Done."
+}
+
+#######################################
+# Parse and set script parameters into associative array.
+# Globals:
+#   parameters: An associative array for script parameters.
+# Arguments:
+#   The whole script command line ($@) where:
+#     parameter keys are prefix with "--"
+#     parameter key and value are separated by with space(s).
+#   Ex.: myscript --parm1 value1 --parm2 value2
+# Outputs:
+#   Writes normal log messages to STDOUT.
+#   Writes error messages to STDERR.
+# Returns:
+#   0 on success
+#   1 on failure
+#######################################
+function utils::parse_parameters() {
+  local readonly KEY_PREFIX="--"
+  local readonly KEY_REGEX_PATTERN="^${KEY_PREFIX}.*$"
+
+  local key
+  local missing_parameter_flag
+  local sorted_keys
+  local unexpected_parameter_flag
+  local usage
+  local value
+
+  utils::echo_action "Mapping input parameter values and checking for unexpected parameters..."
+  unexpected_parameter_flag=false
+  while [[ ${#@} -gt 0 ]]; do
+    key=$1
+    value=$2
+
+    # Test if the parameter key start with the KEY_PREFIX and if the parameter
+    # key without the PARAMETERS_PREFIX is in the expected parameter list.
+    if [[ ${key} =~ $KEY_REGEX_PATTERN && ${parameters[${key:${#KEY_PREFIX}}]+_} ]]; then
+      parameters[${key:${#KEY_PREFIX}}]="${value}"
+    else
+      utils::echo_error "Unexpected parameter: ${key}"
+      unexpected_parameter_flag=true
+    fi
+
+    # Move to the next key/value pair or up to the end of the parameter list.
+    shift $(( 2 < ${#@} ? 2 : ${#@} ))
+  done
+  utils::echo_info "Done."
+
+  utils::echo_action "Checking for missing parameters..."
+  sorted_keys=$(echo ${!parameters[@]} | tr " " "\n" | sort | tr "\n" " ");
+  missing_parameter_flag=false
+  for key in ${sorted_keys}; do
+    if [[ -z ${parameters[${key}]} ]]; then
+      utils::echo_error "Missing parameter: ${key}."
+      missing_parameter_flag=true
+    fi
+  done
+  utils::echo_info "Done."
+
+  # Abort if missing or extra parameters.
+  usage="USAGE: $(basename $0)"
+  if [[ ${unexpected_parameter_flag} == "true" || ${missing_parameter_flag} == "true" ]]; then
+    utils::echo_error "Execution aborted due to missing or extra parameters."
+    for key in ${sorted_keys}; do
+      usage="${usage} -${key} \$${key}"
+    done
+    utils::echo_error "${usage}";
+    exit 1;
+  fi
+
+  utils::echo_action 'Printing input parameter values for debugging purposes...'
```

Comment:
Switching to single quotes, we should keep it consistent.

Reply:
Corrected